lib/spack/spack/solver/concretize.lp

```diff
@@ -161,32 +170,32 @@ node(Dependency) :- node(Package), depends_on(Package, Dependency).
 % dependencies) and get a two-node unconnected graph
 needed(Package) :- root(Package).
 needed(Dependency) :- needed(Package), depends_on(Package, Dependency).
-:- node(Package), not needed(Package),
-   error("All dependencies must be reachable from root").
+unnecessary(Package) :- node(Package), not needed(Package).
```

Comment:
I think the only way this happens right now is if the user asks for `^foo` and `foo` isn't connected to anything in the DAG.  So this is probably the right name.

If it could happen for other reasons it might get weird b/c it implies that the node *is* necessary for a valid solution.

Reply:
I believe everything in this comment is correct.